### ISSUE TITLE
InvalidCleanMethod when using django-allauth

### ISSUE THREAD

**Issue by Tobi-De** [boxed/django-fastdev#23]
Hello, I hope you are well.

There is an issue when using this package with django-allauth.
The [ `LoginForm` defined in django-allauth defines a `clean_login`](https://github.com/pennersr/django-allauth/blob/master/allauth/account/forms.py#L161) method, but since there is no `login` field attached to the form, `django-fastdev` throws an exception which is the right behaviour, but in this case I would like to be able to ignore this exception. Is there a way to ignore exceptions on specific fields?

**Comment by boxed** [comment]
Ah.. interesting. Actually this is a bug in fastdev. It needs to run the validation of `clean_` after the last constructor, but it runs it at the end of the base class constructor, which is clearly wrong. The current code only works for forms without a custom `__init__`!

**Comment by Tobi-De** [comment]
Oh I see, it make sense.

**Comment by Tobi-De** [comment]
I just took a look at the code @boxed 

```python
 orig_form_init = Form.__init__

        def fastdev_form_init(self, *args, **kwargs):
            orig_form_init(self, *args, **kwargs)

            from django.conf import settings
            if settings.DEBUG:
                prefix = 'clean_'
                for name in dir(self):
                    if name.startswith(prefix) and callable(getattr(self, name)) and name[len(prefix):] not in self.fields:
                        fields = '\n    '.join(sorted(self.fields.keys()))

                        raise InvalidCleanMethod(f"""Clean method {name} of class {self.__class__.__name__} won't apply to any field. Available fields:
    {fields}""")

        Form.__init__ = fastdev_form_init
```
The simplest solution I can think of (which seems really bad) would be to have a setting where the user defines a list of their custom forms, then `django-fastev` takes that list and dynamically overrides the `__init__` method. I'm not even sure that would work.

Another solution (perhaps temporary) would be to let users disable validation of the `clean_` methods for forms.

**Comment by boxed** [comment]
I have released 1.7.3 with a different method of doing this clean validation. Please try it out.

**Comment by Tobi-De** [comment]
> I have released 1.7.3 with a different method of doing this clean validation. Please try it out.

Fixed the isssue, thanks
